docs/code-style.md

```diff
@@ -0,0 +1,667 @@
+
+#### Punctuation
+
+Write proper sentences; start with a capital letter and use proper
+punctuation.
+
+#### Top-Level Definitions
+
+Comment every top level function (particularly exported functions),
+and provide a type signature; use Haddock syntax in the comments.
+Comment every exported data type.  Function example:
+
+```haskell
+-- | Send a message on a socket. The socket must be in a connected
+-- state. Returns the number of bytes sent. Applications are
+-- responsible for ensuring that all data has been sent.
+send :: Socket      -- ^ Connected socket
+     -> ByteString  -- ^ Data to send
+     -> IO Int      -- ^ Bytes sent
+```
+
+For functions, the documentation should give enough information to
+apply the function without looking at its definition.
+
+Record example:
+
+```haskell
+-- | Bla bla bla.
+data Person = Person
+    { age  :: !Int     -- ^ Age
+    , name :: !String  -- ^ First name
+    }
+```
+
+For fields that require longer comments, format them this way:
+
+```haskell
+data Record = Record
+    { -- | This is a very very very long comment that is split over
+      -- multiple lines.
+      field1 :: !Text
+
+      -- | This is a second very very very long comment that is split
+      -- over multiple lines.
+    , field2 :: !Int
+    }
+```
+
+#### End-of-Line Comments
+
+Separate end-of-line comments from the code with 2 spaces. Align
+comments for data type definitions. Some examples:
+
+```haskell
+data Parser = Parser
+    !Int         -- Current position
+    !ByteString  -- Remaining input
+
+foo :: Int -> Int
+foo n = salt * n + 9
+  where
+    salt = 453645243  -- Magic hash salt.
+```
+
+#### Links
+
+Use in-line links economically.  You are encouraged to add links for
+API names. It is not necessary to add links for all API names in a
+Haddock comment. We therefore recommend adding a link to an API name
+if:
+
+* The user might actually want to click on it for more information (in
+  your opinion), and
+
+* Only for the first occurrence of each API name in the comment (don't
+  bother repeating a link)
+
+Top-down guideline
+------------------
+
+### LANGUAGE extensions section
+
+Write each `LANGUAGE` pragma on its own line, sort them alphabetically and
+align by max width among them.
+
+```haskell
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE Rank2Types       #-}
+{-# LANGUAGE TemplateHaskell  #-}
+```
+
+### Module name
+
+Use singular when naming modules (e.g. use `Data.Map` and
+`Data.ByteString.Internal` instead of `Data.Maps` and
+`Data.ByteString.Internals`). Sometimes it's acceptable to use plural
+(e. g. `Types`, `Instances`).
+
+### Export Lists
+
+Format export lists as follows:
+
+```haskell
+module Data.Set
+       ( -- * The @Set@ type
+         Set
+       , empty
+       , singleton
+
+         -- * Querying
+       , member
+       ) where
+```
+
+Some clarifications:
+
+1. Use 7 spaces indentation for export list (so that bracket is below
+   the first letter in module name).
```

Comment:
I still do not understand how do people achieve 7 spaces indentation in their editors :crying_cat_face:

Reply:
I use space button ))

Reply:
:why: